src/main/java/com/intuit/graphql/orchestrator/schema/transform/KeyTransformer.java

```diff
@@ -24,20 +28,24 @@
 
   @Override
   public XtextGraph transform(final XtextGraph source) {
-    List<TypeDefinition> entities = source.getTypes().values().stream()
+    Map<String, TypeDefinition> entities = source.getTypes().values().stream()
             .filter(typeDefinition -> typeContainsDirective(typeDefinition, FEDERATION_KEY_DIRECTIVE))
-            .collect(Collectors.toList());
+            .collect(Collectors.toMap(TypeDefinition::getName, Function.identity()));
 
-    for (final TypeDefinition typeDefinition : entities) {
-      for (final Directive directive : typeDefinition.getDirectives()) {
+    if(entities.size() > 0 && source.getServiceProvider().getSeviceType() != ServiceProvider.ServiceType.FEDERATION_SUBGRAPH) {
```

Comment:
i suggest we encapsulate `source.getServiceProvider().getSeviceType() != ServiceProvider.ServiceType.FEDERATION_SUBGRAPH)` to `source.isFederationSubgraph()`

Reply:
I may change it to isFederatedSource just to keep it generic if we have to support other federation implementations in the future